.github/workflows/build-and-release.yml

```diff
@@ -40,33 +45,54 @@ jobs:
           ./run-tests.sh
 
       - name: Coveralls
-        uses: AndreMiras/coveralls-python-action@develop
+        uses: AndreMiras/coveralls-python-action@v20201129
         with:
           parallel: true
           github-token: ${{ secrets.github_token }}
           flag-name: run-${{ matrix.test_number }}
 
-  Finish:
+  Coveralls:
     needs: Test
     runs-on: ubuntu-latest
     steps:
       - name: Coveralls Finished
-        uses: AndreMiras/coveralls-python-action@develop
+        uses: AndreMiras/coveralls-python-action@v20201129
         with:
           github-token: ${{ secrets.github_token }}
           parallel-finished: true
 
   Release:
+    if: ${{ github.event_name == 'push' }}
     needs: Test
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          python-version: 2.7
+          fetch-depth: 0
+
+      - name: Set up Python 2.7
+        uses: actions/setup-python@v2
+        with:
+          python-version: 2.7
+
+      - name: Cache Python Dependencies
+        uses: actions/cache@master
+        id: cache
+        with:
+          path: ${{ env.pythonLocation }}
+          key: ${{ runner.os }}-pip-2.7-${{ github.sha }}
+
+      - name: Install python dependencies
```

Comment:
do we have to install deps twice?

Reply:
We are actually not installing them twice. This step is in "Release" which is separate job. every job is running separately on new docker/VM. So we have to install them there also. Good thing is that there is a step which is caching whole python env on python2 test run and thanks to that this step takes ~9 seconds just to check if all packages are actually installed. It's there in case that there is something wrong with cache. Cache itself it's from the same build, and cache between builds won't be re-used as cache key is generated from commit SHA which should be unique enough to not re-use cache between different PRs